gdx/src/com/badlogic/gdx/graphics/g2d/Sprite.java

```diff
@@ -374,7 +374,8 @@ public void setColor (float r, float g, float b, float a) {
 	 * @see #setColor(Color)
 	 * @see Color#toFloatBits() */
 	public void setPackedColor (float packedColor) {
-		if (this.packedColor != packedColor) {
+		// Handle 0f/-0f special case
+		if (packedColor != this.packedColor || (packedColor  == 0f && this.packedColor == 0f && Float.floatToIntBits(packedColor) != Float.floatToIntBits(this.packedColor))) {
```

Comment:
I like this a lot; it limits the time spent on bit conversions to the rare both-zero-ish state, and should still be accurate in all cases.